### PR TITLE
[GEP-19] Introduce `prometheus-operator` in garden and seed clusters

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity monitoring
/kind enhancement

**What this PR does / why we need it**:
This PR
- adds a Golang component for `prometheus-operator` (based on https://github.com/prometheus-operator/prometheus-operator/blob/main/bundle.yaml)
- extends the `Garden` controller part of `gardener-operator` to deploy it
- extends the `Seed` controller part of `gardenlet` to deploy it (only in case the seed is not the garden cluster at the same time (otherwise, `gardener-operator` already takes over))

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/9065

**Special notes for your reviewer**:
/cc @timuthy @oliver-goetz 
FYI @istvanballok @rickardsjp 

**Release note**:

```feature developer
The `prometheus-operator` (and its related `CustomResourceDefinition`s) are now deployed by default to garden clusters (by `gardener-operator`) and to seed clusters (by `gardenlet`). In the future, it will take over management of the Prometheus and Alertmanager instances.
```

### CODE REVIEW

File: pkg/component/monitoring/prometheusoperator/deployment.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+						RunAsUser:      pointer.Int64(65532),
+						SeccompProfile: &corev1.SeccompProfile{Type: corev1.SeccompProfileTypeRuntimeDefault},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:            "prometheus-operator",
+							Image:           p.values.Image,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Args: []string{
+								fmt.Sprintf("--prometheus-config-reloader=%s", p.values.ImageConfigReloader),
```
Comment (oliver-goetz) [inline pkg/component/monitoring/prometheusoperator/deployment.go:64]: The deployment in `bundle.yaml` you referred to in the PR description includes the argument `--kubelet-service=kube-system/kubelet` too.
I don't really know if we need prometheus-operator to manage this service or not. I just want to ensure that it was left out by purpose and not by accident 😄

**Review by oliver-goetz — COMMENTED** [review]
Nice PR thanks 🚀 

Looks good, except the failing operator e2e test 😅

File: pkg/component/monitoring/prometheusoperator/deployment.go (hunk 1 above)
Comment (rfranzke) [reply pkg/component/monitoring/prometheusoperator/deployment.go:64]: Thanks, yes, I've seen that, but omitted it on purpose for now. I might reintroduce it when working on the migration of the respective Prometheis if needed, though. :)

**Comment by oliver-goetz** [comment]
/lgtm cancel
/approve

@istvanballok @rickardsjp please check and lgtm if the PR suits you 😄

**Review by istvanballok — APPROVED** [review]
Kudos and thank you! :)
/lgtm

**Pull request merged**